From 0c19b2dda95cd6ffd062f409fce953fa1e88ae0a Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:55 -0400
Subject: [usb] xhci: use usb_ss_max_streams in xhci_check_streams_endpoint

Message-id: <1410887463-20674-284-git-send-email-dzickus@redhat.com>
Patchwork-id: 95178
O-Subject: [RHEL7 PATCH 283/591] xhci: use usb_ss_max_streams in xhci_check_streams_endpoint
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a3901538611f8d5180df44f5b61293e70903958f
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Oct 4 17:05:55 2013 +0200

    xhci: use usb_ss_max_streams in xhci_check_streams_endpoint

    The ss_ep_comp bmAttributes filed can contain more info then just the
    streams, use usb_ss_max_streams to properly get max streams.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 38c077a..8d0bb87 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -2971,7 +2971,7 @@ static int xhci_check_streams_endpoint(struct xhci_hcd *xhci,
  ret = xhci_check_args(xhci_to_hcd(xhci), udev, ep, 1, true, __func__);
  if (ret <= 0)
   return -EINVAL;
- if (ep->ss_ep_comp.bmAttributes == 0) {
+ if (usb_ss_max_streams(&ep->ss_ep_comp) == 0) {
   xhci_warn(xhci, "WARN: SuperSpeed Endpoint Companion"
     " descriptor for ep 0x%x does not support streams\n",
     ep->desc.bEndpointAddress);
-- 
1.7.1